shell/src/state.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <string>
+#include <unordered_map>
+
+namespace shell {
+
+class IState {
```

Comment:
What does this class do?

Reply:
For now - storing ENV and probably current_dir (for pwd command). 
For future -  can add history, aliases, etc

Reply:
I mean, do we really need empty virtual class?

Reply:
It is stub since there is no realization for substitution and vars setting.
Alternative is to implement as an abstract class, for me interfaces looks prettier.

Reply:
My general consideration is: we should describe an interfaces for all logical units, and connect all of them using only interfaces. It makes code compliant with SOLID

Also it is great to consider (at least theoretically) how the bash features outside our current requirements can be implemented, and build architecture according with this.
E.g.: streams forwarding (>, >>, &>), subshells, scripts execution, shell-inside-shell